types/index.d.ts

```diff
@@ -128,6 +127,19 @@ export interface ItemDiscount {
   amount?: number;
 }
 
+export interface SwellError {
+  code: string;
+  message: string;
+}
+
+export interface ServerError {
+  error: SwellError;
+}
+
+export interface ErrorResponse {
+  errors: Record<string, SwellError | undefined>;
+}
```

Comment:
It might be possible to narrow down the type further, assuming that the object will only ever include keys from the endpoint's subject type:
```suggestion
export interface ErrorResponse<TSubject> {
  errors: Partial<Record<keyof TSubject, SwellError | undefined>>;
}
```

Reply:
I don't think that this is always the case that the key maps to a specific field.